httpx/http_service.go

```diff
@@ -0,0 +1,165 @@
+package httpx
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/remind101/pkg/retry"
+
+	"golang.org/x/net/context"
+)
+
+type RoundTripper interface {
+	RoundTrip(*http.Request, context.Context) (*http.Response, error)
```

Comment:
The general convention when using context.Context is to make it the first parameter:

```go
RoundTrip(context.Context, *http.Request) (*http.Response, error)
```